unit_tests/test_ssh.py

```diff
@@ -0,0 +1,66 @@
+import unittest
+from pathlib import Path
+import shutil
+import stat
+import tempfile
+
+from archive_auth_mirror.ssh import (
+    add_authorized_key,
+    create_key,
+)
+
+
+class CreateKeyTest(unittest.TestCase):
+
+    def setUp(self):
+        self.tempdir = Path(tempfile.mkdtemp())
+        self.addCleanup(shutil.rmtree, str(self.tempdir))
```

Comment:
I think I'll leave it as it is, since the methods they are testing don't have any charm-logic in them, so using CharmTest just to get a temporary directory seems overkill